R/tags.R

```diff
@@ -1057,9 +1060,28 @@ HTML <- function(text, ..., .noWS = NULL) {
 #'   )
 #' )
 #'
+#' # Setting .noWS for all tags in withTags()
+#' withTags(
+#'   div(
+#'     class = "myclass",
+#'     h3("header"),
+#'     p("One", strong(span("two")), "three")
+#'   ),
+#'   .noWS = c("outside", "inside")
+#' )
+#'
 #'
 #' @export
-withTags <- function(code) {
+withTags <- function(code, .noWS = NULL) {
+  if (!is.null(.noWS)) {
+    .noWS_global <- .noWS
+    tags <- lapply(tags, function(tag) {
+      function(..., .noWS = rlang::missing_arg()) {
+        if (rlang::is_missing(.noWS)) .noWS <- .noWS_global
+        tag(..., .noWS = .noWS)
+      }
+    })
```

Comment:
I believe you can set the default argument right away to avoid using `missing_arg()`.

```suggestion
    .noWSWithTags <- .noWS
    tags <- lapply(tags, function(tag) {
      function(..., .noWS = .noWSWithTags) {
        tag(..., .noWS = .noWS)
      }
    })
```

Reply:
(`.noWS` supplied to `withTags()` is not needed to be `force()`ed as the if is null check forces the value)